engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/RecordScreen.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.mainMenu;
+
+import org.codehaus.plexus.util.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.config.Config;
+import org.terasology.engine.GameEngine;
+import org.terasology.engine.TerasologyConstants;
+import org.terasology.engine.modes.StateLoading;
+import org.terasology.engine.paths.PathManager;
+import org.terasology.game.GameManifest;
+import org.terasology.i18n.TranslationSystem;
+import org.terasology.naming.Name;
+import org.terasology.naming.NameVersion;
+import org.terasology.network.NetworkMode;
+import org.terasology.recording.RecordAndReplayStatus;
+import org.terasology.recording.RecordAndReplayUtils;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.texture.AWTTextureFormat;
+import org.terasology.rendering.assets.texture.Texture;
+import org.terasology.rendering.assets.texture.TextureData;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.animation.MenuAnimationSystems;
+import org.terasology.rendering.nui.layers.mainMenu.savedGames.GameInfo;
+import org.terasology.rendering.nui.layers.mainMenu.savedGames.GameProvider;
+import org.terasology.rendering.nui.widgets.UIButton;
+import org.terasology.rendering.nui.widgets.UIImage;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.rendering.nui.widgets.UIList;
+import org.terasology.utilities.Assets;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+
+public class RecordScreen extends CoreScreenLayer {
```

Comment:
I wonder if there'd be a way to avoid duplication here - if you end up with a unique new screen here, for instance, and multi-world also changes the way previewing works, the two projects would diverge. If instead you find a way to just hook in and add your extras to the existing system then it might merge together OK

Not necessarily high priority right now, but something to consider before aiming to merge the PR